streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -375,150 +199,117 @@ private RuntimeException unAssignChangeLogPartitions() {
-        log.warn("{} Producer of task {} fenced; closing zombie task", logPrefix, task.id());
-        try {
-            task.close(false);
-        } catch (final Exception e) {
-            log.warn("{} Failed to close zombie task due to {}, ignore and proceed", logPrefix, e);
-        }
-        activeTasks.remove(task.id());
-    }
-
-
-    RuntimeException performOnActiveTasks(final TaskAction action) {
-        return performOnTasks(action, activeTasks, "stream task");
-    }
-
-    RuntimeException performOnStandbyTasks(final TaskAction action) {
-        return performOnTasks(action, standbyTasks, "standby task");
-    }
-
-    private RuntimeException performOnTasks(final TaskAction action, final Map<TaskId, Task> tasks, final String taskType) {
-        RuntimeException firstException = null;
-        final Iterator<Map.Entry<TaskId, Task>> it = tasks.entrySet().iterator();
-        while (it.hasNext()) {
-            final Task task = it.next().getValue();
-            try {
-                action.apply(task);
-            } catch (final ProducerFencedException e) {
-                closeZombieTask(task);
-                it.remove();
-            } catch (final RuntimeException t) {
-                log.error("{} Failed to {} " + taskType + " {} due to the following error:",
-                          logPrefix,
-                          action.name(),
-                          task.id(),
-                          t);
-                if (firstException == null) {
-                    firstException = t;
-                }
-            }
-        }
-
-        return firstException;
-    }
-
-
-
     void shutdown(final boolean clean) {
         log.debug("{} Shutting down all active tasks {}, standby tasks {}, suspended tasks {}, and suspended standby tasks {}",
-                  logPrefix, activeTasks.keySet(), standbyTasks.keySet(),
-                  suspendedTasks.keySet(), suspendedStandbyTasks.keySet());
-
-        for (final Task task : allTasks()) {
-            try {
-                task.close(clean);
-            } catch (final RuntimeException e) {
-                log.error("{} Failed while closing {} {} due to the following error:",
-                          logPrefix,
-                          task.getClass().getSimpleName(),
-                          task.id(),
-                          e);
-            }
-        }
+                  logPrefix, active.runningTaskIds(), standby.runningTaskIds(),
+                  active.previousTaskIds(), standby.previousTaskIds());
+
+        active.close(clean);
+        standby.close(clean);
         try {
             threadMetadataProvider.close();
         } catch (final Throwable e) {
             log.error("{} Failed to close KafkaStreamClient due to the following error:", logPrefix, e);
         }
         // remove the changelog partitions from restore consumer
         unAssignChangeLogPartitions();
+        taskCreator.close();
```

Comment:
For line 215 above: is it intentional that we do not check the returned exception any more in `unAssignChangeLogPartitions`?